stan/math/gpu/multiply.hpp

```diff
@@ -104,10 +104,15 @@ inline auto multiply(const matrix_gpu& A, const matrix_gpu& B) {
  * @throw <code>std::invalid_argument</code> if the
  *   number of columns in A and rows in B do not match
  */
-template <typename T1, typename T2>
-inline matrix_gpu operator*(const T1& A, const T2& B) {
+inline matrix_gpu operator*(const matrix_gpu& A, const matrix_gpu& B) {
   return multiply(A, B);
 }
+inline matrix_gpu operator*(const matrix_gpu& A, const double scalar) {
+  return multiply(A, scalar);
+}
+inline matrix_gpu operator*(const double scalar, const matrix_gpu& B) {
+  return multiply(scalar, B);
```

Comment:
You should probably just make this also be `multiply(B, scalar)` to avoid confusion.